amazonsqs/amazonsqs_utils.bal

```diff
@@ -0,0 +1,107 @@
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import ballerina/crypto;
+import ballerina/encoding;
+import ballerina/http;
+import ballerina/system;
+import ballerina/time;
+import ballerina/log;
+
+# Handles the HTTP response.
+#
+# + httpResponse - Http response or error
+# + return - If successful returns `json` response. Else returns error.
+function handleResponse(http:Response|error httpResponse) returns json|error {
+    if (httpResponse is http:Response) {
+        if (httpResponse.statusCode == http:NO_CONTENT_204){
+            //If status 204, then no response body. So returns json boolean true.
+            return true;
+        }
+        var xmlResponse = httpResponse.getXmlPayload();
+        if (xmlResponse is xml) {
+            var jsonResponse = xmlResponse.toJSON({ preserveNamespaces: false });
+            if (httpResponse.statusCode == http:OK_200) {
+                //If status is 200, request is successful. Returns resulting payload.
+                return jsonResponse;
+            } else {
+                //If status is not 200 or 204, request is unsuccessful. Returns error.
+                string errorMsg = STATUS_CODE + COLON_SYMBOL + jsonResponse["error"].code.toString()
+                    + SEMICOLON_SYMBOL + WHITE_SPACE + MESSAGE + COLON_SYMBOL + WHITE_SPACE
+                    + jsonResponse["error"]["message"].toString();
+                error err = error(AMAZONSQS_ERROR_CODE, { message: errorMsg });
+                return err;
+            }
+        } else {
+                error err = error(AMAZONSQS_ERROR_CODE, { message: "Response payload is not XML" });
+                return err;
+        }
+    } else {
+        error err = error(AMAZONSQS_ERROR_CODE, { message: "Error occurred while invoking the REST API" });
+        return err;
+    }
+}
+
+function generatePOSTRequest(string accessKeyId, string secretAccessKey, string host, string amzTarget, string canonicalUri,
+                        string region, string payload) returns http:Request|error {
+    time:Time time = check time:toTimeZone(time:currentTime(), "GMT");
+    string amzDate = check time:format(time, ISO8601_BASIC_DATE_FORMAT);
+    string dateStamp = check time:format(time, SHORT_DATE_FORMAT);
+    string contentType = "application/x-www-form-urlencoded";
+    string requestParameters =  payload;
+    string canonicalQuerystring = "";
+    string canonicalHeaders = "content-type:" + contentType + "\n" + "host:" + host + "\n" + "x-amz-date:" + amzDate + "\n" + "x-amz-target:" + amzTarget + "\n";
```

Comment:
Shall we define "\n" and "/" as constants? 
Also, other Strings like "AWS4-HMAC-SHA256" can define as constants.

Reply:
No need as it would reduce the readability.